packages/artifact/src/internal/upload/upload-zip-specification.ts

```diff
@@ -75,10 +81,11 @@ export function getUploadZipSpecification(
           - file3.txt
   */
   for (let file of filesToZip) {
-    if (!fs.existsSync(file)) {
+    const stats = fs.lstatSync(file, {throwIfNoEntry: false})
```

Comment:
mildly important part: we switched from `statSync` to [`lstatSync`](https://nodejs.org/api/fs.html#fs_fs_lstatsync_path_options)

> similar to `stat` unless `path` refers to a symbolic link, in which case the link itself is stat-ed, not the file that it refers to

we'll use these stats to check [`isSymbolicLink`](https://nodejs.org/api/fs.html#fs_stats_issymboliclink) when we add the file to the zip, and resolve the target's contents.